### PR TITLE
Issue#350 Body Parts Validation

### PR DESCRIPTION
**Pull request by MrNiceUK**
#350 
Also fixes the bug where for arms (and related parts), it generally wouldn't update correctly when the only difference between the parts is a material override.

### CODE REVIEW

**Comment by MrNiceUK** [comment]
Added the Helmet #354 fix, and renumbered arm related override material comparison fixes from #350 to #354

**Comment by Musashi1584** [comment]
Can this be potentially merged? If so why hasn't it already been merged?

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XGCharacterGenerator.uc [hunk 1]
```diff
@@ -658,9 +658,11 @@ function SetArmsLegsAndDeco(X2SimpleBodyPartFilter BodyPartFilter)
 		RandomizeSetBodyPart(PartTemplateManager, kSoldier.kAppearance.nmRightArm, "RightArm", BodyPartFilter.FilterByTorsoAndArmorMatch);
 		RandomizeSetBodyPart(PartTemplateManager, kSoldier.kAppearance.nmLeftArmDeco, "LeftArmDeco", BodyPartFilter.FilterByTorsoAndArmorMatch);
 		RandomizeSetBodyPart(PartTemplateManager, kSoldier.kAppearance.nmRightArmDeco, "RightArmDeco", BodyPartFilter.FilterByTorsoAndArmorMatch);
-		RandomizeSetBodyPart(PartTemplateManager, kSoldier.kAppearance.nmLeftForearm, "LeftForearm", BodyPartFilter.FilterByTorsoAndArmorMatch);
-		RandomizeSetBodyPart(PartTemplateManager, kSoldier.kAppearance.nmRightForearm, "RightForearm", BodyPartFilter.FilterByTorsoAndArmorMatch);
+	//Begin Issue #350
 	}
+	RandomizeSetBodyPart(PartTemplateManager, kSoldier.kAppearance.nmLeftForearm, "LeftForearm", BodyPartFilter.FilterByTorsoAndArmorMatch);
+	RandomizeSetBodyPart(PartTemplateManager, kSoldier.kAppearance.nmRightForearm, "RightForearm", BodyPartFilter.FilterByTorsoAndArmorMatch);
+	//End Issue #350
```
Comment (MrNiceUK) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XGCharacterGenerator.uc]: These two lines were moved out of the if(kSoldier.kAppearance.nmArms == '') block. PR #390 also wrapped them with a forced appearance check, which it also did for the "Hero Deco" slots below. The "correct" resolution to this apparent conflict, is to simply put these lines within the same forced appearance check block as the "Hero Deco's" below

Comment (MrNiceUK) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XGCharacterGenerator.uc]: These two lines were moved out of the if(`kSoldier.kAppearance.nmArms == ''`) block. PR #390 also wrapped them with a forced appearance check, which it also did for the "Hero Deco" slots below. The "correct" resolution to this apparent conflict, is to simply put these lines within the same forced appearance check block as the "Hero Deco's" below

**Comment by MrNiceUK** [comment]
Made a comment over in the changes..

**Pull request closed without merging**